internal/errs2/group.go

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package errs2
+
+import "sync"
+
+// Group is a collection of goroutines working on subtasks that are part of
+// the same overall task.
+type Group struct {
+	wg      sync.WaitGroup
+	errOnce sync.Once
+	errors  []error
+}
+
+// Go calls the given function in a new goroutine.
+func (group *Group) Go(f func() error) {
+	group.wg.Add(1)
+
+	go func() {
+		defer group.wg.Done()
+
+		if err := f(); err != nil {
+			group.errors = append(group.errors, err)
```

Comment:
this is a data race

Reply:
ah, sorry, I should focus on one thing at once